src/components/views/rooms/PinnedEventsPanel.js

```diff
@@ -0,0 +1,171 @@
+
+        return (
+            <div className="mx_PinnedEventTile">
+                <div className="mx_PinnedEventTile_actions">
+                    <AccessibleButton className="mx_PinnedEventTile_gotoButton mx_textButton" onClick={this.onTileClicked}>
+                        { _t("Jump to message") }
+                    </AccessibleButton>
+                    { unpinButton }
+                </div>
+
+                <MemberAvatar member={sender} width={avatarSize} height={avatarSize} />
+                <span className="mx_PinnedEventTile_sender">
+                    {sender.name}
+                </span>
+                <MessageEvent mxEvent={this.props.mxEvent} className="mx_PinnedEventTile_body" />
+            </div>
+        );
+    }
+});
+
+module.exports = React.createClass({
+    displayName: 'PinnedEventsPanel',
+    propTypes: {
+        // The Room from the js-sdk we're going to show pinned events for
+        room: React.PropTypes.object.isRequired,
+
+        onCancelClick: React.PropTypes.func,
+    },
+
+    getInitialState: function() {
+        return {
+            loading: true,
+        };
+    },
+
+    componentDidMount: function() {
+        this._updatePinnedMessages();
+    },
+
+    _updatePinnedMessages: function() {
+        const pinnedEvents = this.props.room.currentState.getStateEvents("m.room.pinned_events", "");
+        if (!pinnedEvents || !pinnedEvents.getContent().pinned) {
+            this.setState({ loading: false, pinned: [] });
+        } else {
+            const promises = [];
+            const cli = MatrixClientPeg.get();
+
+            pinnedEvents.getContent().pinned.map(eventId => {
+                promises.push(cli.getEventTimeline(this.props.room.getUnfilteredTimelineSet(), eventId, 0).then(timeline => {
+                    const event = timeline.getEvents().find(e => e.getId() === eventId);
+                    return {eventId, timeline, event};
+                }).catch(err => {
+                    console.error("Error looking up pinned event " + eventId + " in room " + this.props.room.roomId);
+                    console.error(err);
+                    return null; // return lack of context to avoid unhandled errors
+                }));
+            });
+
+            Promise.all(promises).then(contexts => {
+                // Filter out the messages before we try to render them
+                const pinned = contexts.filter(context => {
+                    if (!context) return false; // no context == not applicable for the room
+                    if (context.event.getType() !== "m.room.message") return false;
+                    if (context.event.isRedacted()) return false;
+                    return true;
+                });
+
+                this.setState({ loading: false, pinned });
+            });
+        }
+    },
+
+    _getPinnedTiles: function() {
+        if (this.state.pinned.length == 0) {
+            return <div>{ _t("No pinned messages.") }</div>;
```

Comment:
Would we ever show this? Surely if there are no pinned messages, you'd just show nothing at the top?

Reply:
The pinned messages pane is a drawer that is dismissed by default. This is more to communicate to people that there's nothing to see when they pop it open.